### PR TITLE
feat: add searching mechanism for videos

### PR DESCRIPTION
**Pull request by TheMadNerd**
#80

### CODE REVIEW

**Review by PatrykKuniczak — APPROVED** [review]
LGTM

File: react-app/src/pages/popup/components/video/section/article/video-article.tsx [hunk 1]
```diff
@@ -21,10 +20,26 @@ const VideoArticleStyles = styled(Stack)<TComponentTag>(({ theme }) =>
...
+  views,
+  title,
+  isLoading,
+}: {
+  thumbnail: string;
+  authorName: string;
+  avatar: string;
+  whenPublished: string;
+  views: string;
+  title: string;
```
Comment (PatrykKuniczak) [inline react-app/src/pages/popup/components/video/section/article/video-article.tsx:37]: Create interface for that

File: react-app/src/pages/popup/components/video/section/article/authorInfo/author-info.tsx [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 import { StyledSkeleton } from '@pages/popup/components/shared/styled-skeleton';
 
-export const StyledAuthorInfo = ({ isLoading }: { isLoading: boolean }) => (
+export const StyledAuthorInfo = ({
+  avatar,
+  authorName,
+  isLoading,
+}: {
+  avatar: string;
+  authorName: string;
```
Comment (PatrykKuniczak) [inline react-app/src/pages/popup/components/video/section/article/authorInfo/author-info.tsx:12]: use here a part of this new interface

File: react-app/src/pages/popup/components/video/section/article/videoInfo/video-infos.tsx [hunk 3]
```diff
@@ -3,23 +3,27 @@ import { StyledVideoInfo } from '@pages/popup/components/video/section/article/v
...
-export const StyledVideoInfos = ({ isLoading }: { isLoading: boolean }) => (
+export const StyledVideoInfos = ({
+  whenPublished,
+  views,
+  title,
+  isLoading,
+}: {
+  whenPublished: string;
+  views: string;
+  title: string;
```
Comment (PatrykKuniczak) [inline react-app/src/pages/popup/components/video/section/article/videoInfo/video-infos.tsx:14]: also use a part of new interface

File: react-app/src/pages/popup/components/video/section/article/videoInfo/video-infos.tsx [hunk 4]
```diff
@@ -3,23 +3,27 @@ import { StyledVideoInfo } from '@pages/popup/components/video/section/article/v
...
 import { StyledSkeleton } from '@pages/popup/components/shared/styled-skeleton';
 
-export const StyledVideoInfos = ({ isLoading }: { isLoading: boolean }) => (
+export const StyledVideoInfos = ({
+  whenPublished,
+  views,
+  title,
+  isLoading,
+}: {
+  whenPublished: string;
```
Comment (PatrykKuniczak) [inline react-app/src/pages/popup/components/video/section/article/videoInfo/video-infos.tsx:12]: this prop should be 'publishedAt'

File: react-app/src/pages/popup/routes/videos.route.tsx [hunk 5]
```diff
@@ -13,12 +46,18 @@ export const VideosRoute = () => {
 
   return (
     <StyledVideosSection>
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
+      {filteredVideos?.map(({ id, thumbnail, avatar, authorName, whenPublished, views, title }) => (
```
Comment (PatrykKuniczak) [inline react-app/src/pages/popup/routes/videos.route.tsx:49]: in map destructure only id, and use it for 'key' but the restprops use on the component below, like key={id} ...restProps

**Review by PatrykKuniczak — CHANGES_REQUESTED** [review]

File: react-app/src/pages/popup/routes/videos.route.tsx [hunk 6]
```diff
@@ -13,12 +46,9 @@ export const VideosRoute = () => {
...
     <StyledVideosSection>
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
+      {filteredVideos?.map(({ id, ...restProps }) => (
+        <StyledVideoArticle key={id} {...restProps} isLoading={isLoading} />
+      ))}
```
Comment (masterHAWK99) [inline react-app/src/pages/popup/routes/videos.route.tsx:51]: Skeletons won't show up when `filteredVideos` will be undefined. You should show skeletons when the whole list of videos is loading, not each one.

File: react-app/src/pages/popup/routes/videos.route.tsx [hunk 7]
```diff
@@ -1,24 +1,65 @@
...
-    <StyledVideosSection>
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-      <StyledVideoArticle isLoading={isLoading} />
-    </StyledVideosSection>
-  );
+  if (isLoading === true) {
```
Comment (masterHAWK99) [inline react-app/src/pages/popup/routes/videos.route.tsx:51]:
```suggestion
  if (isLoading) {
```

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]

File: react-app/src/pages/popup/components/video/section/article/videoInfo/video-infos.tsx [hunk 8]
```diff
@@ -2,29 +2,23 @@ import { Stack } from '@mui/system';
...
 import { StyledVideoTitle } from '@pages/popup/components/video/section/article/videoInfo/video-title';
 import { useTranslation } from 'react-i18next';
-import { StyledSkeleton } from '@pages/popup/components/shared/styled-skeleton';
+import { IVideo } from '@interfaces';
 
-export const StyledVideoInfos = ({ isLoading }: { isLoading: boolean }) => {
+export const StyledVideoInfos = ({ publishedAt, views, title }: Pick<IVideo, 'publishedAt' | 'views' | 'title'>) => {
   const { t } = useTranslation();
 
+  const publishedDate = new Date(publishedAt);
```
Comment (PatrykKuniczak) [inline react-app/src/pages/popup/components/video/section/article/videoInfo/video-infos.tsx:10]: that's unnecessary, put this prop directly below

File: react-app/src/pages/popup/routes/videos.route.tsx [hunk 9]
```diff
@@ -1,24 +1,65 @@
+import useSearch from '@hooks/use-search';
 import { StyledVideoArticle } from '@pages/popup/components/video/section/article/video-article';
 import { StyledVideosSection } from '@pages/popup/components/video/section/videos-section';
-import { useEffect } from 'react';
+import { useDeferredValue, useEffect, useMemo } from 'react';
+import thumbnail from '@assets/img/thumbnail.png';
+import { StyledVideoArticleSkeleton } from '@pages/popup/components/video/section/skeleton/video-article-skeleton';
 
 export const VideosRoute = () => {
+  const videos = useMemo(
```
Comment (PatrykKuniczak) [inline react-app/src/pages/popup/routes/videos.route.tsx:9]: why memo, for static values?

**Review by PatrykKuniczak — CHANGES_REQUESTED** [review]

File: react-app/src/pages/popup/routes/videos.route.tsx (hunk 9 above)
Comment (TheMadNerd) [reply react-app/src/pages/popup/routes/videos.route.tsx:9]: without memo this error pops out
"The 'videos' function makes the dependencies of useMemo Hook (at line 39) change on every render. Move it inside the useMemo callback. Alternatively, wrap the definition of 'videos' in its own useCallback() Hook."

Comment (PatrykKuniczak) [reply react-app/src/pages/popup/routes/videos.route.tsx:9]: below also you shouldn't use memo, but leave it here, we will change it in the future

**Review by PatrykKuniczak — APPROVED** [review]
LGTM

**Review by masterHAWK99 — APPROVED** [review]

**Pull request merged**